src/permissions/extractors.rs

```diff
@@ -2,27 +2,40 @@ use actix_web::dev::ServiceRequest;
 use actix_web::Error;
 use std::future::Future;
 
-pub trait PermissionsExtractor<'a> {
+pub trait PermissionsExtractor<'a, Req> {
     type Future: Future<Output = Result<Vec<String>, Error>>;
 
-    fn extract(&self, request: &'a ServiceRequest) -> Self::Future;
+    fn extract(&self, request: &'a mut ServiceRequest) -> Self::Future;
```

Comment:
In fact, this `trait` has lost backward compatibility. 
But all its implementations support both versions (mut / non-mut ref).

So, it should be invisible to the end user.
But there are concerns that there might be users with custom implementation of this `trait`